Assets/00_Content/Scripts/Player/PlayerPickUp.cs

```diff
@@ -0,0 +1,90 @@
+using Interactables;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+namespace Player {
+
+	public class PlayerPickUp : MonoBehaviour {
+
+		[SerializeField] private Transform playerGrabTransform;
+
+		//Should probably be displayed on a canvas instead.
+		[SerializeField] private GameObject highlight;
+
+		private List<PickUp> pickUps;
+		private PickUp closestPickUp;
+		private PickUp pickedUpItem;
+
+		private void Start() {
+
+			pickUps = new List<PickUp>();
+		}
+
+		public void PickUpClosestItem() {
+			HighlightPickUp(null);
+
+			if (pickUps.Count < 1)
+				return;
+
+			closestPickUp.PickUpItem(playerGrabTransform);
+			pickUps.Remove(closestPickUp);
+			pickedUpItem = closestPickUp;
+		}
+
+		public void DropItem() {
+
+			if (pickedUpItem == null)
+				return;
+
+			pickedUpItem.DropItem();
+			pickedUpItem = null;
+		}
+
+		public void UseItem() {
+			if (pickedUpItem != null && pickedUpItem is IUseable)
+				Debug.Log("Using item...");
+		}
+
+		private void OnTriggerEnter(Collider other) {
+
+			if (other.gameObject.CompareTag("PickUp") && pickedUpItem == null) {
+
+				pickUps.Add(other.gameObject.GetComponent<PickUp>());
+				closestPickUp = pickUps.OrderBy(x => Vector3.Distance(x.transform.position, transform.position)).First();
```

Comment:
```suggestion
				closestPickUp = pickUps.OrderBy(x => (x.transform.position - transform.position).sqrMagnitude).First();
```
You could optimize this by sorting over square magnitude instead of using the absolute distance. Same on line 72.

Also, using a for loop to find the closest pick-up would be faster because OrderBy has to allocate a new conainer and do a full sort.